frontend/src/components/dashboard/settings/api-key/ApiKey.tsx

```diff
@@ -1,155 +1,127 @@
+      setIsPageLoading(false)
+    })()
+  }, [])
 
-    const timeoutId = setTimeout(() => {
-      setApiKeyState(ApiKeyState.COPY)
-
-      if (!apiKeyRef.current) return
-      apiKeyRef.current.blur()
-    }, RESET_COPY_TIMEOUT)
-
-    return () => clearTimeout(timeoutId)
-  }, [apiKeyState])
+  const modalContext = useContext(ModalContext)
 
-  async function onButtonClick() {
-    switch (apiKeyState) {
-      case ApiKeyState.GENERATE:
-      case ApiKeyState.REGENERATE:
-        if (apiKey) {
-          modalContext.setModalContent(
-            <ConfirmModal
-              title="Are you sure?"
-              subtitle="Generating a new API key will revoke your current one."
-              buttonText="Confirm"
-              onConfirm={onGenerateConfirm}
-            />
-          )
-        } else {
-          await onGenerateConfirm()
-        }
-        break
-      case ApiKeyState.COPY:
-        if (!apiKeyRef.current) return
-        apiKeyRef.current.select()
-        document.execCommand('copy')
-        setApiKeyState(ApiKeyState.COPIED)
-        break
-      default:
-        break
-    }
+  function openCopyModal(apiKey: ApiKeyType) {
+    modalContext.setModalContent(<CopyModal apiKey={apiKey} />)
   }
 
-  async function onGenerateConfirm() {
-    setErrorMsg('')
-    try {
-      const newApiKey = await regenerateApiKey()
-      setApiKey(newApiKey)
-      setApiKeyState(ApiKeyState.COPY)
-    } catch (e) {
-      setErrorMsg((e as Error).message)
-    }
-    if (onGenerate) onGenerate()
+  function openGenerateModal() {
+    modalContext.setModalContent(
+      <CreateModal
+        onSuccess={(apiKey: ApiKeyType) => {
+          addApiKey(apiKey)
+          openCopyModal(apiKey)
+        }}
+      />
+    )
   }
 
-  let buttonLabel = ''
-  let buttonIcon = ''
-  let buttonClass = ''
-  switch (apiKeyState) {
-    case ApiKeyState.GENERATE:
-      buttonLabel = 'Generate'
-      buttonIcon = 'bx-key'
-      buttonClass = styles.greenButton
-      break
-    case ApiKeyState.COPY:
-      buttonLabel = 'Copy'
-      buttonIcon = 'bx-copy'
-      buttonClass = styles.greenButton
-      break
-    case ApiKeyState.COPIED:
-      buttonLabel = 'Copied'
-      buttonIcon = 'bx-check'
-      buttonClass = styles.blueButton
-      break
-    case ApiKeyState.REGENERATE:
-      buttonLabel = 'Regenerate'
-      buttonIcon = 'bx-refresh'
-      buttonClass = styles.blueButton
-      break
-    default:
-      break
+  function openDeleteModal(keyId: string) {
+    const apiKey = apiKeys.find((k) => k.id === keyId)
+    if (!apiKey) return
+    modalContext.setModalContent(
+      <ConfirmModal
+        title={`Are you sure you want to delete API key "${apiKey.label}"?`}
+        subtitle="Deleting your API key is irreversible, and any integrations using this key will need to be updated with a new API key to continue working."
+        onConfirm={() => {
+          removeApiKey(keyId)
+          void deleteApiKey(keyId)
```

Comment:
if deleteApiKey api call fails, UI will still show that API key was removed which might be misleading, perhaps we should put removeApiKey after a successful deleteApiKey call